new function to check the software environment (Matlab or Octave)

becker89 opened the pull request:
*random seed initaliziation needs to be done differently in Octave

markbangert commented:
@lucasburigo could you maybe also have a quick look at this to make sure we are not running in the wring direction with regard to Octave support? It is our intention to have the matRad main release run out of the box on Octave upon the next release before xMas.

burigolucas commented:
We are in the write direction to have better Octave support. However, there are still a few points that need to be addressed to improve Octave support, namely:

- `clearvars` is not available in Octave, the analogous would be `clear`. Maybe simply include the OCTAVE/MATLAB switch to handle that (cf [matRad_fluenceOptimization.m](https://github.com/e0404/matRad/blob/dev/matRad_fluenceOptimization.m#L203), [matRad_directApertureOptimization.m](https://github.com/e0404/matRad/blob/dev/matRad_directApertureOptimization.m#L114), [dicomImport/matRad_importDicom.m](https://github.com/e0404/matRad/blob/dev/dicomImport/matRad_importDicom.m#L157))
- the test `exist('griddedInterpolant','class')` (cf [matRad_interp1.m](https://github.com/e0404/matRad/blob/dev/matRad_interp1.m#L39), [matRad_calcPhotonDose.m](https://github.com/e0404/matRad/blob/dev/matRad_calcPhotonDose.m#L257) and [matRad_calcPhotonDose.m](https://github.com/e0404/matRad/blob/dev/matRad_calcPhotonDose.m#L298)) issues a warning as the test for type class is not implemented in Octave yet. If there is no penalty on efficiency, we could change the test to `exist('griddedInterpolant')`. In any case, `griddedInterpolant` is not available in Octave.
- `uitable` is not implemented in Octave yet (cf [matRad_showQualityIndicators.m](https://github.com/e0404/matRad/blob/dev/matRad_showQualityIndicators.m#L43))
- `gobjects` is not implemente in Octave yet (cf [plotting/matRad_plotIsoDoseLines.m](https://github.com/e0404/matRad/blob/dev/plotting/matRad_plotIsoDoseLines.m#L75), [plotting/matRad_plotVoiContourSlice.m](https://github.com/e0404/matRad/blob/dev/plotting/matRad_plotVoiContourSlice.m#L54), [plotting/matRad_plotIsoDoseLines3D.m](https://github.com/e0404/matRad/blob/dev/plotting/matRad_plotIsoDoseLines3D.m#L60))
- the cubic interpolation method for interp3 (cf [tools/matRad_gammaIndex.m](https://github.com/e0404/matRad/blob/dev/tools/matRad_gammaIndex.m#L122)) is not implemented yet in Octave.

becker89 commented:
Thanks for the input Lucas. I will update the pull request accordingly. Maybe you can run the example(s) one more time before we merge. I will update you when I have made the changes.

becker89 commented:
@burigolucas can you check it the updated pull request works on octave ?

burigolucas commented on tools/matRad_gammaIndex.m:
```diff
@@ -119,7 +122,12 @@
+        case 'MATLAB'
+            cubex2 = interp3(cubex2,n,'cubic');
+        case 'OCTAVE'
+            warning('3-D interpolation not yet supported by octave. Function will continue without interpolation.')
```
@becker89: 3-D interpolation with interp3 is supported in Octave, only the method 'cubic' that is not (see [https://www.gnu.org/software/octave/doc/v4.2.0/Multi_002ddimensional-Interpolation.html#XREFinterp3](https://www.gnu.org/software/octave/doc/v4.2.0/Multi_002ddimensional-Interpolation.html#XREFinterp3)).

becker89 commented:
I have changed the cubic interpolation in the gamma index function for octave to the default method (linear)

burigolucas commented:
A few other points:

- the test `exist('griddedInterpolant','class')` (see [matRad_calcPhotonDose.m](https://github.com/e0404/matRad/blob/dev/matRad_calcPhotonDose.m#L257) and [matRad_calcPhotonDose.m](https://github.com/e0404/matRad/blob/dev/matRad_calcPhotonDose.m#L298)) issues a warning as the test for type class is not implemented in Octave yet.
- add "plotting" to path in examples/matRad_example0_photons.m
- add "plotting" and "tools" to path in examples/matRad_example3_protons.m
- change histogram to hist in examples/matRad_example3_protons.m for Octave as histogram is not implemented in Octave yet.

becker89 commented:
Hi Lucas, thanks for the update. I have replaced the check for the griddedInterpolant with a check isMatlab/isOctave. In case of Matlab the griddetInterpolant class will be used (supported since Matlab 2011) and if the code is exectued from Octave, then interp2 is used. Further, I have added missing subdirectories ('plotting','tools') in affected code sections.
